src/service/crypto-verifier-impl.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev} from '../log';
+import {getService} from '../service';
+import {
+  encodeByteArray,
+} from
+  '../../third_party/closure-library/encode-byte-array-trampoline-generated';
```

Comment:
I will leave this question to others. The original code used utilities in goog.crypt. There was a way to import such things into AMP, so I used it.

Just copying the goog.crypt code seemed wrong. Rewriting seemed silly / gratuitous. I agree that the current process for using these routines is awkward and error-prone.

You tell me what to do.